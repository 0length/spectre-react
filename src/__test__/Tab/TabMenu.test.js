import React from 'react'
import { shallow, render } from 'enzyme'

import { TabMenu } from '../..'

describe('TabMenu', () => {
  it('Should render default .tab markup', () => {
    const wrapper = shallow(<TabMenu />)

    expect(wrapper.type()).toBe('ul')
    expect(wrapper.hasClass('tab')).toBe(true)
  })

  it('Should render children', () => {
    const wrapper = shallow(
      <TabMenu>
        <p>hello world</p>
      </TabMenu>
    )

    expect(wrapper.children().length).toBe(1)
    expect(wrapper.find('p').length).toBe(1)
  })

  it('Should pass classnames', () => {
    const wrapper = shallow(<TabMenu className='extra' />)

    expect(wrapper.hasClass('extra')).toBe(true)
    expect(wrapper.hasClass('tab')).toBe(true)
  })

  it('Should use tab-block class', () => {
    const wrapper = shallow(<TabMenu block />)

    expect(wrapper.hasClass('tab-block')).toBe(true)
  })

  it('Should have TabMenu.Item component', () => {
    expect(TabMenu.Item).toBeDefined()
    expect(typeof TabMenu.Item).toBe('function')
  })

  it('Should render as a custom element', () => {
    const wrapper = shallow(<TabMenu renderAs='div'>custom</TabMenu>)

    expect(wrapper.text()).toBe('custom')
    expect(wrapper.hasClass('tab')).toBe(true)
    expect(wrapper.type()).toBe('div')
  })

  it('Should match the snapshot', () => {
    const wrapper = shallow(<TabMenu />)

    expect(wrapper).toMatchSnapshot()
  })

  it('Should match the use case snapshot', () => {
    const wrapper = render(
      <TabMenu block>
        <TabMenu.Item
          active
          onClick={() => window.alert('clicked 1')}
        >
          <a className='btn btn-link'>Active</a>
        </TabMenu.Item>
        <TabMenu.Item
          onClick={() => window.alert('clicked 2')}
        >
          <a className='btn btn-link'>Another Tab</a>
        </TabMenu.Item>
      </TabMenu>
    )

    expect(wrapper).toMatchSnapshot()
  })
})
