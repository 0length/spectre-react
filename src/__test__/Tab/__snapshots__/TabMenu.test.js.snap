// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabMenu Should match the snapshot 1`] = `
<ul
  className="tab"
/>
`;

exports[`TabMenu Should match the use case snapshot 1`] = `
<ul
  class="tab tab-block"
>
  <li
    class="tab-item active"
  >
    <a
      class="btn btn-link"
    >
      Active
    </a>
  </li>
  <li
    class="tab-item"
  >
    <a
      class="btn btn-link"
    >
      Another Tab
    </a>
  </li>
</ul>
`;
